content/posts/gsoc-physiopy/index.md

```diff
@@ -2,6 +2,104 @@
 title = 'GSoC Physiopy'
 date = 2024-09-03T19:33:41+03:00
 draft = false
+toc = true
 +++
 
-Google Summer of Code
\ No newline at end of file
+During the summer of 2024, I had the chance to become a Google Summer of Code student, contributing in the [Physiopy](https://physiopy.github.io/) community, under the [INCF](https://www.incf.org/) organization. In this page, which is also serving as my Work Product Submission for finalizing my GSoC participation, the project will be presented, how it progressed during the summer and what future directions it could take. 
+Emphasis is going to be given to the development breakdown, the rationale of all the contributions and the main ideas for future development.
+
+## 1. Project Overview
+{{< lead >}}
+*"Semi-Automated Workflows for Physiological Signals"*
+{{< /lead >}}
+
+In general, the project entailed the upgrade and restructuring of existing Physiopy packages, `peakdet` and `phys2denoise`, in order to be used more harmoniously with each other. 
+Following the upgrade, the development of semi-automated workflows was planned, along with the accompanying CLI. 
+
+## 2. Pull Request Synopsis
+### `peakdet`
+- https://github.com/physiopy/peakdet/pull/67
+- https://github.com/physiopy/peakdet/pull/62
+### `phys2denoise`
+- https://github.com/physiopy/phys2denoise/pull/54
+- https://github.com/physiopy/phys2denoise/pull/56
+- https://github.com/physiopy/phys2denoise/pull/57
+### `physutils`
+- https://github.com/physiopy/physutils/pull/2
+- https://github.com/physiopy/physutils/pull/4
+- https://github.com/physiopy/physutils/pull/7
+
+## 3. Development Overview
+
+### 3.1 Infrastructure 
+
+### 3.2 Optimizing peakdet/phys2denoise interoperability
+
+The `peakdet` and `phys2denoise` packages, have been both very critical packages of the physiopy suite, with their operation being frequently sequential in multiple fMRI denoising applications. 
+Meaning, that in many common data processing routines, the physiological data first needed to pass through `peakdet`, for filtering and peak/trough detection, and then through `phys2denoise` for the computation of the several denoising metrics.
+However, by nature up to this point, `peakdet` was built in an "object-oriented" fashion, being structured around the `Physio` object, while `phys2denoise` was predominantly "function-oriented".
+The main idea was to incorporate the `Physio` object into both packages, to exploit some of its powerful features, such as operation history management and history "replaying" and easy interfacing with multiple input physiological data types.
+Therefore, the main challenge was to make this migration, incorporating these features to `phys2denoise` and adding more to cater to the additional needs, while continuing to support its "function-only" use.
+
+Initially, prior to the upgrade, a new package was issued for this scope: `physutils`.
```

Comment:
I think it's good to tie this statement to a timeline (initially is vague, something like "at the beginning of my GSoC project" gives us more context that you were the one leading this).